lib/rules/padding-line-between-statements.js

```diff
@@ -537,40 +563,46 @@ module.exports = {
                     match(nextNode, configure.next);
 
                 if (matched) {
-                    return PaddingTypes[configure.blankLine];
+                    if (typeof configure.blankLine === "number") {
+                        return [configure.blankLine, PaddingTypes.always];
+                    }
+                    return [0, PaddingTypes[configure.blankLine]];
                 }
             }
-            return PaddingTypes.any;
+            return [0, PaddingTypes.any];
         }
 
         /**
          * Gets padding line sequences between the given 2 statements.
          * Comments are separators of the padding line sequences.
          * @param {ASTNode} prevNode The previous statement to count.
          * @param {ASTNode} nextNode The current statement to count.
-         * @returns {Array<Token[]>} The array of token pairs.
+         * @param {number} blankLine Allowed number of blank lines.
+         * @returns {[number, Array<Token[]>]} The number of blank lines found and array of token pairs.
          * @private
          */
-        function getPaddingLineSequences(prevNode, nextNode) {
+        function getPaddingLineSequences(prevNode, nextNode, blankLine) {
```

Comment:
Can we update the description of this function?

If I got it right, this function now returns `[foundBlankLine, pairs]`, where:

* `foundBlankLine` is number of blank lines in the longest sequence of consecutive blank lines.
* `pairs` is array of token pairs having
  * any number of blank lines between if `blankLine === 0`
  * exactly `blankLine` blank lines between if `blankLine !== 0`


Or maybe better consider splitting this into multiple functions (e.g. the old `getPaddingLineSequences` + one or two additional functions that would work on `pairs` returned from `getPaddingLineSequences`)?

Reply:
@mdjermanovic If you add the function separately, there will be another extra loop. I suggest updating the description, like this:
```
        /**
         * Number of blank lines in the longest sequence of consecutive blank lines
         * @typedef {number} FoundBlankLine
         * @private
         */
        /**
         * Types of blank lines, `any`, `never`, and `always`
         *  - any number of blank lines between if blankLine === 0
         *  - exactly blankLine blank lines between if blankLine !== 0
         *  - or specified by the user: `any`, `never`, and `always`
         * @see {@link PaddingTypes}
         * @typedef {Object} PaddingType
         * @private
         */
        /**
         * Finds the last matched configure from configureList.
         * @param {ASTNode} prevNode The previous statement to match.
         * @param {ASTNode} nextNode The current statement to match.
         * @returns {[FoundBlankLine, PaddingType]} The allowed number of blank lines and tester of the last matched configure.
         * @private
         */
```

Reply:
@mdjermanovic If you add a separate function, there will most likely be another extra loop.
I offer a variant of the description:
```

        /**
         * Expected number of blank lines in the longest sequence of consecutive blank lines
         * @typedef {number} ExpectedBlankLine
         * @private
         */
        /**
         * Types of blank lines, `any`, `never`, and `always`
         *  - `always` if specified a number of blank lines
         *  - or specified key: `any`, `never`, and `always`
         * @see {@link PaddingTypes}
         * @typedef {Object} PaddingType
         * @private
         */
        /**
         * Finds the last matched configure from configureList.
         * @param {ASTNode} prevNode The previous statement to match.
         * @param {ASTNode} nextNode The current statement to match.
         * @returns {[ExpectedBlankLine, PaddingType]} The allowed number of blank lines and tester of the last matched configure.
         * @private
         */
        function getPaddingType(prevNode, nextNode) {
...
```
        /**
         * Number of blank lines in the longest sequence of consecutive blank lines
         * @typedef {number} FoundBlankLine
         * @private
         */
        /**
         * Array of token pairs having:
         * - any number of blank lines between if blankLine === 0
         * - exactly blankLine blank lines between if blankLine !== 0
         * @typedef {Array<Token[]>} Pairs
         * @private
         */
        /**
         * Gets padding line sequences between the given 2 statements.
         * Comments are separators of the padding line sequences.
         * @param {ASTNode} prevNode The previous statement to count.
         * @param {ASTNode} nextNode The current statement to count.
         * @param {number} blankLine Allowed number of blank lines.
         * @returns {[FoundBlankLine, Pairs]} The number of blank lines found and array of token pairs.
         * @private
         */
        function getPaddingLineSequences(prevNode, nextNode, blankLine) {
...
```

```

Reply:
I personally think this function is doing too much, and because of that requires such a complex description.

One or two additional iterations over `pairs` (which will by far most often have `0` or `1` elements, rarely `2` or more) shouldn't be a performance issue.

This would also eliminate redundant parameter `foundBlankLine` in `function verifyForAlways`. It can be calculated from `paddingLines`, when needed.